roles/matrix-synapse/templates/synapse/homeserver.yaml.j2

```diff
@@ -1039,6 +1039,27 @@ media_store_path: "/matrix-media-store-parent/{{ matrix_synapse_media_store_dire
 #    config:
 #       directory: /mnt/some/other/directory
 
+# Matrix S3 Storage Provider
+{% if matrix_synapse_s3_media_store_enabled %}
+media_storage_providers:
```

Comment:
A better way to inject this configuration is to do it how we do for Synapse modules (see `modules:` in `roles/matrix-synapse/templates/synapse/homeserver.yaml.j2` and the `matrix_synapse_modules` list variable: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/dd83d88cb04791a19f343fd3750f974a3a19c64b/roles/matrix-synapse/templates/synapse/homeserver.yaml.j2#L24

Certain tasks in the playbook may inject modules into the `matrix_synapse_modules` at runtime, and then the whole `matrix_synapse_modules` list is just spewed out into the configuration.

The benefit of this is that users can manipulate the `matrix_synapse_modules` (and in our case `matrix_synapse_media_storage_providers`) list manually as well. If they enable some ready-made media storage provider (`matrix_synapse_media_storage_provider_s3_enabled`) it will get injected into their list without overwriting it.

To see how some module injects itself into the list, see how shared-secret-auth does it: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/dd83d88cb04791a19f343fd3750f974a3a19c64b/roles/matrix-synapse/tasks/ext/shared-secret-auth/setup_install.yml#L26-L37